app/src/main/java/com/mealsmadeeasy/data/FakeMealStore.kt

```diff
@@ -91,4 +92,64 @@ class FakeMealStore : MealStore {
             ?: Observable.just(emptyList<Ingredient>())!!
 
 
+    override fun addMealToMealPlan(meal: Meal, date: DateTime, mealPeriod: MealPeriod) {
+        var added = false
+        var changedPlan = mealPlan.value
+        mealPlan.value.meals
+                .filter { it.date.dayOfMonth() == date.dayOfMonth() && it.mealPeriod == mealPeriod }
+                .forEach {
+                    var temp = emptyList<MealPlanEntry>()
+                    mealPlan.value.meals.forEach {
```

Comment:
Okay, I'm having a lot of trouble trying to figure out what this is doing. It might help to not use the default `it` parameter name here since it's overloaded. Also, can this not be done on a `map` or something?